### PR TITLE
CMS - PKCS #7

### PR DESCRIPTION
**Pull request by richardschneider**
A proposal for creating cryptographic protected data using the key chain.  It uses `PKCS #7 aka RFC 5652` to encrypt the data. 

This can be used to implement `ipfs crypt encrypt` and `ipfs crypt decrypt` in [spec keystore](https://github.com/ipfs/specs/tree/master/keystore)

### CODE REVIEW

**Comment by richardschneider** [comment]
This spec is missing a verification and signing.

**Comment by richardschneider** [comment]
Verification requires the keychain to manage public keys.  Does verification fail if a public key is missing for a signer?  Are multiple signers allowed?

**Comment by richardschneider** [comment]
@diasdavid Theses test are randomly failing because `libp2p-crypto` is slow.  Can you review/merge/release https://github.com/libp2p/js-libp2p-crypto/pull/117?

File: .travis.yml [hunk 1]
```diff
@@ -14,15 +14,11 @@ matrix:
...
   - npm run test
-  - npm run coverage
 
 before_script:
   - export DISPLAY=:99.0
   - sh -e /etc/init.d/xvfb start
 
-after_success:
-  - npm run coverage-publish
-
```
Comment (diasdavid) [inline .travis.yml]: Doesn't seem that coverage got added to Circle https://github.com/libp2p/js-libp2p-keychain/blob/master/circle.yml

**Comment by diasdavid** [comment]
@richardschneider updated deps and rebased master onto this branch. Mind checking why there are failures? You need a release of keychain with this PR, right?

**Comment by richardschneider** [comment]
@diasdavid Thanks for the fast response.

No, I just need the current `master` to update its deps and release to NPM.  However, I will try to determine why this failed.

**Comment by richardschneider** [comment]
@diasdavid Its a problem with the latest `deepmerge`, see https://github.com/KyleAMathews/deepmerge/issues/87#issuecomment-349530307.  They do not believe its an problem!

Please rollback to `"deepmerge": "^1.5.2",`

**Comment by diasdavid** [comment]
@richardschneider is this ready as well?

**Comment by richardschneider** [comment]
@diasdavid Yes its RTM.  I was just waiting for other stuff to settle down.

**Pull request merged**